drivers/soundwire/intel.c

```diff
@@ -472,13 +472,12 @@ void sdw_intel_process_wakeen_event(struct sdw_intel_ctx *ctx)
 		 * the wakeen event and let codec driver check codec status
 		 */
 		list_for_each_entry(slave, &bus->slaves, node) {
-			if (slave->prop.wake_capable) {
-				if (slave->status != SDW_SLAVE_ATTACHED &&
-				    slave->status != SDW_SLAVE_ALERT)
-					continue;
-
+			/*
+			 * avoid waking up the non-existence devices defined in
+			 * ACPI table or the devices don't support this feature
+			 */
+			if (slave->dev_num_sticky && slave->prop.wake_capable)
```

Comment:
discard devices that are defined in ACPI tables but not physically present and devices that cannot generate wakes.

Reply:
Thanks! I will refine it.